misk/src/main/kotlin/misk/concurrent/ExecutorServiceModule.kt

```diff
@@ -0,0 +1,34 @@
+package misk.concurrent
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder
+import misk.inject.KAbstractModule
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import kotlin.reflect.KClass
+
+/**
+ */
+class ExecutorServiceModule(
+  private val annotation: KClass<out Annotation>,
+  private val executorService: ExecutorService
+) : KAbstractModule() {
+  override fun configure() {
+    bind<ExecutorService>().annotatedWith(annotation.java).toInstance(executorService)
+  }
+
+  companion object {
+    fun of(annotation: KClass<out Annotation>, nameFormat: String): ExecutorServiceModule {
```

Comment:
Per Effective Java #41, can we name these according to the `Executors` methods? This could be `withCachedThreadPool()` and the other one `withFixedThreadPool()`

Reply:
done :)